src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2528,8 +2532,16 @@ bool RTPSParticipantImpl::is_reader_ignored(
 }
 
 bool RTPSParticipantImpl::ignore_participant(
-        const GuidPrefix_t& /*participant_guid*/)
+        const GuidPrefix_t& participant_guid)
 {
+    if (!is_participant_ignored(participant_guid))
+    {
+        {
+            std::unique_lock<shared_mutex> _(ignored_mtx_);
+            ignored_participants_.insert(participant_guid);
+        }
+        return true;
```

Comment:
We should also remove the participant proxy from PDP, using reason IGNORED_PARTICIPANT

Reply:
The proxy removal is handled outside this method (see [here](https://github.com/eProsima/Fast-DDS/blob/1c3da086cd0745cb838061fcc3b7bcafd717963d/src/cpp/rtps/builtin/discovery/participant/PDPListener.cpp#L169)). This allows the method to be called from inside `remove_remote_participant` upon returning from the discovery callback.